### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by AalexxJjamess**

### CODE REVIEW

File: Highway.cpp [hunk 1]
```diff
@@ -13,17 +18,37 @@ void Highway::changeSpeed(int newSpeed)
...
+        car->closeWindows();
+    }
+    else if (auto* moto = dynamic_cast<Motorcycle*>(v))
+    {
+        moto->lanesplitAndRace(100);
+    }
+    else if (auto* truck = dynamic_cast<SemiTruck*>(v))
+    {
+        //SemiTruck->tryToEvade();
+    }
```
Comment (matkatmusic) [inline Highway.cpp:33]: re-read the instructions.
- [ ] revise

File: Highway.cpp [hunk 2]
```diff
@@ -13,17 +18,37 @@ void Highway::changeSpeed(int newSpeed)
...
+        car->tryToEvade();
+    }
+    else if (auto* moto = dynamic_cast<Motorcycle*>(v))
+    {
+        moto->tryToEvade();
+    }
+    else if (auto* truck = dynamic_cast<SemiTruck*>(v))
+    {
+        //SemiTruck->tryToEvade();
+    }
```
Comment (matkatmusic) [inline Highway.cpp:51]: re-read the instructions.
- [ ] revise

File: HighwayPatrol.cpp [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
 }
 
+HighwayPatrol::~HighwayPatrol() {}
+
 void HighwayPatrol::scanHighway(Highway* h)
 {
     std::cout << name << ": scanning highway for speeders" << std::endl;
 
-    for( size_t i = h->vehicles.size(); --i >= 0; )
+    for( size_t i = h->vehicles.size(); --i != 0; )
```
Comment (matkatmusic) [inline HighwayPatrol.cpp:19]: This should actually be `for( size_t i = h->vehicles.size(); i-- > 0; )` but I didn't update your project before you started it.
- [ ] please revise this line for me.

File: main.cpp [hunk 4]
```diff
@@ -109,15 +115,34 @@ int main()
...
+        highway.addVehicle(&car);
+    }
+    for (Motorcycle& motorbike : motorcycles)
+    {
+        highway.addVehicle(&motorbike);
+    }
+    for (SemiTruck& semi : trucks)
+    {
+        highway.addVehicle(&semi);
+    }
```
Comment (matkatmusic) [inline main.cpp:144]: you should use `auto&` instead of the Type.
- [ ] revise

File: HighwayPatrol.cpp (hunk 3 above)
Comment (AalexxJjamess) [reply HighwayPatrol.cpp:19]: Fixed.

File: Highway.cpp (hunk 1 above)
Comment (AalexxJjamess) [reply Highway.cpp:33]: Added comply with pull over.

File: Highway.cpp (hunk 2 above)
Comment (AalexxJjamess) [reply Highway.cpp:51]: Added comply with pull over.

**Pull request merged**